src/types/config.js

```diff
@@ -87,12 +87,20 @@
  *   URL template for username links
  */
 
+/**
+ * RPC settings derived from `ConfigFromHost['requestConfigFromFrame']`
+ *
+ * @typedef RPCSettings
+ * @prop {Window} targetFrame
+ * @prop {RequestConfigFromFrameOptions['origin']} origin
+ */
+
```

Comment:
There are many, many ways we could do this, and it may evolve soon/quickly based on how implementation proceeds. But the basic goal is to retain, in settings, what we need to make RPC calls _after_ configuration-munging time.